src/cam_touptek.cpp

```diff
@@ -463,6 +463,20 @@ bool CameraToupTek::Connect(const wxString& camIdArg)
     m_cam.SetOption(TOUPCAM_OPTION_DFC, 0);
     m_cam.SetOption(TOUPCAM_OPTION_SHARPENING, 0);
 
+    // check camera Conversion Gain
+    int cgSupported;
+    if (m_cam.GetOption(TOUPCAM_OPTION_CG, &cgSupported))
+    {
+        // camera support Conversion Gain, set mode HCG
+        m_cam.SetOption(TOUPCAM_OPTION_CG, 1);
+        Debug.Write("TOUPTEK: Camera supports Conversion Gain, set to HCG (1)\n");

```

Comment:
the call to SetOption on line 471 could fail (return false); if it fails we should not write this log message indicating that it succeeded.
In other words, better check the result of SetOption() and only write this message if it returned true.
if SetOption fails, it already logs an error message internally so no need to print anything else if SetOption returns false.